pkg/providers/cloudstack/cloudstack.go

```diff
@@ -657,10 +659,34 @@ func (cs *CloudStackTemplateBuilder) GenerateCAPISpecControlPlane(clusterSpec *c
 		buildOption(values)
 	}
 
+	cpCsMtName, ok := values[cpTemplateNameKey]
+	if !ok {
+		return nil, fmt.Errorf("unable to determine control plane template name")
+	}
+	cpMachineTemplate := CloudStackMachineTemplate(fmt.Sprintf("%v", cpCsMtName), cs.controlPlaneMachineSpec, cs.now)
```

Comment:
I would like the machine template name to come from a method, so it's easy to replicate

Reply:
currently, machine template name is only available from the `values` map. All this code is doing is reading from that map and then converting to a string. Are you suggesting factoring out that logic? Or are you proposing we centralize all the ways that the name is generated?

Reply:
I see. But we are in `GenerateCAPISpecControlPlane` method, which returns `[]byte`
Then does this mean we are going to need to duplicate the logic that calls "go template + api structs" in the controller? Just to be able to access the api structs before they get marshaled to yaml

Reply:
Ah I see
Is the machine template name being used in the go template? If not, no need to pass it as part of the map, we can make it a explicit argument to this/other methods

As for conversion, does `fmt.Sprintf("%s", cpCsMtName)` work?

Reply:
Looks like it does, thanks for the tip